source/adios2/engine/hdf5/HDF5ReaderP.h

```diff
@@ -11,4 +11,115 @@
 #ifndef ADIOS2_ENGINE_HDF5_HDF5READERP_H_
 #define ADIOS2_ENGINE_HDF5_HDF5READERP_H_
 
+#include "HDF5Common.h"
+
+#include "adios2/ADIOSMPICommOnly.h"
+#include "adios2/core/Engine.h"
+
+namespace adios
+{
+
+class HDF5Reader : public Engine
+{
+
+public:
+    /**
+     * Constructor for single HDF5 reader engine, reads from HDF5 format
+     * @param name unique name given to the engine
+     * @param accessMode
+     * @param mpiComm
+     * @param method
+     */
+    HDF5Reader(ADIOS &adios, const std::string name,
+               const std::string accessMode, MPI_Comm mpiComm,
+               const Method &method);
+
+    virtual ~HDF5Reader();
+
+    Variable<void> *InquireVariable(const std::string &variableName,
```

Comment:
All of these `InquireVariableTypeName` member functions would be better suited to a single template function with specializations for each type.